applications/flash_attention_v2/collective/xe_flash_attn_mma.hpp

```diff
@@ -300,8 +302,54 @@ struct CollectiveMmaAttention<MainloopIntelPVC<Stages>, TileShape_, ElementQ_, S
     copy(params.gmem_tiled_copy_v, tVgV, tVrV);
     cute::gemm(tiled_mma, accum, tPr, tCrV, frag_src);
   }
+
+  template <bool VarLen, class ProblemShape>
+  CUTLASS_DEVICE static constexpr Params get_updated_copies(Params const& params, ProblemShape const& problem_shape, int const& l_coord) {
+    if constexpr (!VarLen) {
+      return params;
+    } else {
+      auto [batch, num_heads, seq_len_qo, seq_len_kv, head_size_qk, head_size_vo] = problem_shape;
+
+      auto qo_cumulative_length = get<2>(problem_shape).cumulative_length;
+      auto kv_cumulative_length = get<3>(problem_shape).cumulative_length;
+
+      int offset_q = l_coord == 0 ? 0 : num_heads * head_size_qk * qo_cumulative_length[l_coord];
+      int offset_k = l_coord == 0 ? 0 : num_heads * head_size_qk * kv_cumulative_length[l_coord];
+      int offset_v = l_coord == 0 ? 0 : num_heads * head_size_vo * kv_cumulative_length[l_coord];
```

Comment:
What is the value of e.g. `qo_cumulative_length[0]`? I would expect it to be `0` always, in which case these ternaries could be simply replaced by their `false` branches.